backend/server.py

```
@@ -8083,6 +8083,60 @@ async def sam_speak_backend(body: SamSpeakRequest, user: dict = Depends(get_curr
+    identity = str(user.get("email") or user.get("_id") or "commander")
+    room = (body.room or "").strip() or f"{cfg.room_prefix}-{identity.split('@')[0]}"
+    try:
+        token = mint_access_token(
+            api_key=cfg.api_key,
+            api_secret=cfg.api_secret,
+            identity=identity,
+            room=room,
+            name=str(user.get("name") or "Commander"),
+        )
+    except ValueError as exc:
+        raise HTTPException(status_code=500, detail=f"Token minting failed: {exc}")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)